crypto/dsa/dsa_ossl.c

```diff
@@ -367,9 +367,13 @@ static int dsa_do_verify(const unsigned char *dgst, int dgst_len,
         return -1;
     }
 
-    i = BN_num_bits(dsa->params.q);
-    /* fips 186-3 allows only different sizes for q */
-    if (i != 160 && i != 224 && i != 256) {
+    qlen = BN_num_bits(dsa->params.q);
+    /*
+     * SP 800-131Ar2 allows many sizes for q for verification while
+     * FIPS 186-3 only allows 160, 224 or 256.  We check and permit
+     * the valid values from both.
+     */
+    if (qlen < 160 || qlen > 256 || (qlen > 224 && qlen < 256)) {
```

Comment:
Isnt it 160 =< N <= 224 or N == 256?

Reply:
They are the allowed lengths.  This is the opposite check for not allowed lengths.
I.e. the following are not allowed:

- N < 160
- 224 < N < 256
- N > 256

which is what's being checked.

Reply:
Kind of moot if this change is dropped....